src/Symfony/Component/VarDumper/Caster/SocketCaster.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\VarDumper\Caster;
+
+use Symfony\Component\VarDumper\Cloner\Stub;
+
+/**
+ * @author Alexandre Daubois <alex.daubois@gmail.com>
+ */
+class SocketCaster
+{
+    public static function castSocket(\Socket $h, array $a, Stub $stub, bool $isNested): array
+    {
+        socket_getsockname($h, $addr, $port);
+        $info = stream_get_meta_data(socket_export_stream($h));
+
+        $a += [
+            Caster::PREFIX_VIRTUAL.'address' => $addr,
+            Caster::PREFIX_VIRTUAL.'port' => $port,
+            Caster::PREFIX_VIRTUAL.'info' => new EnumStub(array_intersect_key(
+                $info,
+                [
+                    'timed_out' => new ConstStub($info['timed_out'] ? 'true' : 'false'),
+                    'blocked' => new ConstStub($info['blocked'] ? 'true' : 'false'),
+                    'eof' => new ConstStub($info['eof'] ? 'true' : 'false'),
+                    'unread_bytes' => new ScalarStub($info['unread_bytes']),
+                    'stream_type' => new ConstStub($info['stream_type']),
+                    'wrapper_type' => new ConstStub($info['wrapper_type'] ?? ''),
+                    'wrapper_data' => new ConstStub($info['wrapper_data'] ?? ''),
+                    'mode' => new ConstStub($info['mode']),
+                    'seekable' => new ConstStub($info['seekable'] ? 'true' : 'false'),
+                    'uri' => new ConstStub($info['uri'] ?? ''),
```

Comment:
none of the ConstStub here make sense
Also, are all entries useful?
From php-src, not all of them are populated IIUC:
https://github.com/php/php-src/blob/2473f57ba7c0ceeb6c08dfe335995925d0497028/ext/sockets/sockets.c#L2386